test/cri-containerd/container_test.go

```diff
@@ -288,10 +290,7 @@ func Test_RunContainer_VirtualDevice_GPU_LCOW(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 	defer cancel()
 	containerID := createContainer(t, client, ctx, containerRequest)
-	defer removeContainer(t, client, ctx, containerID)
```

Comment:
It would be better to leave the `defer` calls in place so that we clean up properly in the case of a panic. If we need the `defer` to occur earlier, we could move code out to a separate function (or just use an anonymous function in-line).

Reply:
I think there should be two tests then. One to check that the device actually exists in the container and a separate one that checks that the vhd was removed, since these two things require different handling of the container's lifetime. But in that case, I really feel like there should just be a test for scsi add/remove that checks that a device no longer exists on the uvm after container clean up.

Reply:
I'm fine with having a test just for SCSI add/remove, and not also validating that here.

Reply:
that's what I'm thinkin :)